### PR TITLE
[CS-3204] New Request Payment design

### PR DESCRIPTION
**Pull request by oidoug**
### Description

Overhaul of Payment Request flow with new design, while also removing deprecated ShowQRCodeModal and improving the readability of locally used components.

- [x] Completes #(CS-3204)

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

https://user-images.githubusercontent.com/129619/155027288-20596db7-f658-4fab-aef2-ca5556650d1a.mp4

### CODE REVIEW

**Comment by kdowlin** [comment]
@oidoug this looks good - how long is the QR load on device?

**Comment by oidoug** [comment]
> @oidoug this looks good - how long is the QR load on device?

@kdowlin The video is from my "modern" Samsung, so around a couple of seconds.
On my older iPhone 6, it takes around 4 secs to load.

File: cardstack/src/components/QRCode/StyledQRCode.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  const [isFocus, setFocus] = useState(false);
+  const [isQRCodeRendering, setQRCodeState] = useState(true);
+  const onQRCodeRendered = useCallback(() => setQRCodeState(false), []);
+  useFocusEffect(
+    useCallback(() => {
+      setFocus(true);
+
+      return () => setFocus(false);
+    }, [])
+  );
```
Comment (danibonilha) [inline cardstack/src/components/QRCode/StyledQRCode.tsx:27]: Wondering if we still need this piece, now that is not used on a new screen anymore, it seems if we remove it, it loads faster, you can reference this [PR discussion](https://github.com/cardstack/cardwallet/pull/576) to decide if we need to keep this

**Review by danibonilha — APPROVED** [review]
LGTM, just an small suggestion about an improvement

File: cardstack/src/components/QRCode/StyledQRCode.tsx (hunk 1 above)
Comment (oidoug) [reply cardstack/src/components/QRCode/StyledQRCode.tsx:27]: It's still valid, same problem and resolution. I think it takes some time to instantiate something in EasyQRCode, so doing that after transitioning screens looks a bit better.

Hopefully, we'll be able to use `qrcode-svg` and that can be removed.

Comment (topwebtek7) [reply cardstack/src/components/QRCode/StyledQRCode.tsx:27]: `qrcode-svg` is not supported in RN yet.

**Review by topwebtek7 — APPROVED** [review]
LGTM! 👍🏼

File: cardstack/src/components/QRCode/StyledQRCode.tsx (hunk 1 above)
Comment (oidoug) [reply cardstack/src/components/QRCode/StyledQRCode.tsx:27]: Ah, sorry, I meant to say `react-native-qrcode-svg`. Theres a test up using it in pr #660

**Pull request merged**